File: 905-sort-array-by-parity/905-sort-array-by-parity.swift
class Solution {
    func sortArrayByParity(_ nums: [Int]) -> [Int] {
        var dummyNums = nums
        var i = 0
        var j = dummyNums.count - 1
        while i < j{
            if dummyNums[i] % 2 == 0{
                i += 1
            }
            else{
                if dummyNums[j] % 2 != 0{
                    j -= 1
                }else{
                    dummyNums.swapAt(i, j)
                    i += 1
                    j -= 1
                }
            }
        }
        return dummyNums
    }
}

File: 905-sort-array-by-parity/905-sort-array-by-parity.go
func sortArrayByParity(nums []int) []int {
    var (
        i = 0
        j = len(nums) - 1
    )
    for i < j{
        if nums[i] % 2 == 0{
            i += 1
        }else{
            if nums[j] % 2 != 0{
                j -= 1
            }else{
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j -= 1
            }
        }
    }
    return nums
}